GoTeamApp/GoTeamApp/LabelTableViewCell.swift

```diff
@@ -1,14 +1,61 @@
 import UIKit
 
+protocol LabelTableViewCellDelegate {
+    func editButtonAction(cell: LabelTableViewCell)
+    func filterTasksForSelectedLabelAction(cell: LabelTableViewCell)
+    func deleteLabelCell(sender : LabelTableViewCell)
+}
+
 class LabelTableViewCell: UITableViewCell {
 
     @IBOutlet weak var labelName: UILabel!
+    @IBOutlet weak var editLabelButton: UIButton!
+    @IBOutlet weak var tasksFilterButton: UIButton!
+    
+    
+    @IBOutlet weak var deleteView: UIView!
+    @IBOutlet weak var topView: UIView!
+    var delegate: LabelTableViewCellDelegate?
+    
+    var tableCellUtil : TableCellUtil!
+    
+    var label : Labels? {
+        didSet {
+            cellFGViewLeadingSpaceConstraint.constant = 0
+            if let label = label {
+                self.labelName.text = label.labelName
+            }
+        }
+    }
+    
+    @IBOutlet weak var cellFGViewLeadingSpaceConstraint: NSLayoutConstraint!
+    
+    @IBAction func editButtonAction(_ sender: Any) {
+        if let _ = delegate {
+            delegate?.editButtonAction(cell: self)
```

Comment:
FYI, not strictly necessary to have the let _ = delegate check prior to this call, since a nil delegate here would basically do nothing. This is a case where you have employed optional chaining, so you can expect it to fail gracefully
https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/OptionalChaining.html